resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ResteasyClientBuilderImpl.java

```diff
@@ -317,6 +317,11 @@ public ResteasyProviderFactory getProviderFactory()
          // create a new one
          providerFactory = new LocalResteasyProviderFactory(ResteasyProviderFactory.newInstance());
          RegisterBuiltin.register(providerFactory);
+
+         if (ResteasyProviderFactory.peekInstance() != null)
+         {
+            providerFactory.initializeClientProviders(ResteasyProviderFactory.getInstance());
+         }
       }

```

Comment:
Hi @xstefank,

This looks good, but, now that I see it, I'm wondering why we didn't just build in the first place providerFactory as
```
providerFactory = new LocalResteasyProviderFactory(ResteasyProviderFactory.getInstance());
```
That would also handle, for example, clientMessageBodyReaders and clientMessageBodyWriters.

-Ron

Reply:
Hi @ronsigal, this would take into account all filters including server ones which I think is unnecessary for client provider factory. I would also like to add all other client providers - [RESTEASY-2163](https://issues.jboss.org/browse/RESTEASY-2163)

Reply:
Yeah, @xstefank, good point.